ROCm error: ggml-cuda.cu:6246: invalid device function

Running inference with 
./main -ngl 1 -m llama-2-13b-chat.Q5_K_M.gguf -p "write hello"
an error appears:
Log start
main: build = 1268 (bedb92b)
main: built with AMD clang version 17.0.0 (https://github.com/RadeonOpenCompute/llvm-project roc-5.7.0 23352 d1e13c532a947d0cbfc94759c00dcf152294aa13) for x86_64-unknown-linux-gnu
main: seed  = 1695481826
ggml_init_cublas: found 1 ROCm devices:
  Device 0: , compute capability 11.0

... <some log messages> ...

CUDA error 98 at /code/llama.cpp/ggml-cuda.cu:6246: invalid device function

running with Ubuntu 22, rocm 5.7, AMD Radeon 7900 XTX in docker environment

with run command:

docker run -it --network=host --device=/dev/kfd --device=/dev/dri --ipc=host --shm-size 24G --group-add video --cap-add=SYS_PTRACE --security-opt seccomp=unconfined

Building from cmake without forcing CC/CXX fails entirely since cmake is trying to build with my regular system gcc (which is why I assume this project's ``README.md`` forces CC/CXX as well when using cmake with hipBLAS):

```
[  1%] Built target BUILD_INFO
[  2%] Building CXX object CMakeFiles/ggml-rocm.dir/ggml-cuda.cu.o
c++: error: unrecognized command-line option ‘--offload-arch=gfx900’
c++: error: unrecognized command-line option ‘--offload-arch=gfx906’
c++: error: unrecognized command-line option ‘--offload-arch=gfx908’
c++: error: unrecognized command-line option ‘--offload-arch=gfx90a’
c++: error: unrecognized command-line option ‘--offload-arch=gfx1030’
make[2]: *** [CMakeFiles/ggml-rocm.dir/build.make:76: CMakeFiles/ggml-rocm.dir/ggml-cuda.cu.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:538: CMakeFiles/ggml-rocm.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

Looking over the Makefile, it seems like it specifically compiles ``ggml-cuda`` with clang included within ROCm if ``LLAMA_HIPBLAS`` is set and for the rest it uses the default system compiler (which seems to be gcc for me):

https://github.com/ggerganov/llama.cpp/blob/c091cdfb24621710c617ea85c92fcd347d0bf340/Makefile#L404-L424

However it does use ``$(ROCM_PATH)/bin/hipcc`` instead of ``$(ROCM_PATH)/llvm/bin/clang`` / ``$(ROCM_PATH)/llvm/bin/clang++``. Not sure what the difference is though. Unfortunately I also don't really know how to replicate that behavior with cmake. When I try only forcing ``CXX`` to the same compiler used in the makefile for ``ggml-cuda`` (``CXX=/opt/rocm/bin/hipcc cmake .. -DLLAMA_HIPBLAS=ON``), the build does go through and executing ``main`` now reports ``main: built with cc (GCC) 13.2.1 20230801 for x86_64-pc-linux-gnu``, however it still ends with the same ``CUDA error 98``.

--- a/flake.nix
+++ b/flake.nix
@@ -11,8 +11,7 @@
         meta.mainProgram = "llama";
         inherit (pkgs.stdenv) isAarch32 isAarch64 isDarwin;
         buildInputs = with pkgs; [ openmpi ];
-        osSpecific = with pkgs; buildInputs ++
-        (
+        osSpecific = with pkgs; buildInputs ++ (
           if isAarch64 && isDarwin then
             with pkgs.darwin.apple_sdk_11_0.frameworks; [
               Accelerate
@@ -96,12 +95,15 @@
         };
         packages.rocm = pkgs.stdenv.mkDerivation {
           inherit name src meta postPatch nativeBuildInputs postInstall;
-          buildInputs = with pkgs; buildInputs ++ [ hip hipblas rocblas ];
+          buildInputs = with pkgs.rocmPackages; buildInputs ++ [ clr hipblas rocblas ];
           cmakeFlags = cmakeFlags ++ [
             "-DLLAMA_HIPBLAS=1"
             "-DCMAKE_C_COMPILER=hipcc"
             "-DCMAKE_CXX_COMPILER=hipcc"
-            "-DCMAKE_POSITION_INDEPENDENT_CODE=ON"
+            # Build all targets supported by rocBLAS. When updating search for TARGET_LIST_ROCM
+            # in github.com/ROCmSoftwarePlatform/rocBLAS/blob/develop/CMakeLists.txt
+            # and select the line that matches the current nixpkgs version of rocBLAS.
+            "-DAMDGPU_TARGETS=gfx803;gfx900;gfx906:xnack-;gfx908:xnack-;gfx90a:xnack+;gfx90a:xnack-;gfx940;gfx941;gfx942;gfx1010;gfx1012;gfx1030;gfx1100;gfx1101;gfx1102"
           ];
         };
         apps.llama-server = {

